src/colvar_UIestimator.h

```diff
@@ -0,0 +1,831 @@
+#ifndef COLVAR_UIESTIMATOR_H
+#define COLVAR_UIESTIMATOR_H
+
+#include <cmath>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include <typeinfo>
+
+// only for colvar module!
+// when integrated into other code, just remove this line and "...cvm::backup_file(...)"
+#include "colvarmodule.h"
+
+namespace UIestimator
+{
+    const int Y_SIZE = 21;
+    const int HALF_Y_SIZE = 10;
+    const int EXTENDED_X_SIZE = HALF_Y_SIZE;
+
+    class n_matrix    // spare matrix, stores the distribution matrix of n(x,y)
+    {
+    public:
+        n_matrix() {}
+        n_matrix(const std::vector<double> & lowerboundary,   // lowerboundary of x
+            const std::vector<double> & upperboundary,   // upperboundary of 
+            const std::vector<double> & width,           // width of x
+            const int y_size)           // size of y, for example, ysize=7, then when x=1, the distribution of y in [-2,4] is considered
+        {
+            this->lowerboundary = lowerboundary;
+            this->upperboundary = upperboundary;
+            this->width = width;
+            this->dimension = lowerboundary.size();
+            this->y_size = y_size;     // keep in mind the internal (spare) matrix is stored in diagonal form
+            this->y_total_size = int(pow(y_size, dimension) + 0.000001);
```

Comment:
These "magic small numbers" are not very clean. To get a rounded rather than truncated value, use 0.5 (see the colvargrid class).

Reply:
This is a trick to avoid something like int(4.999999999) == 4. I'm not sure if I should use 0.5 here.

Reply:
@fhh2626 What about 4.999999998?  Since you haven't looked at the colvarscript class, here is the snippet for your convenience:
```
int nbins_round = (int)(nbins+0.5);
```
This is due to the integer cast working by truncation.

Reply:
@giacomofiorin Thanks for your reply. In this sentence, a value of 0.5 also works because 
`pow(y_size, dimension)`
should be an integer.
But in
`loop_flag_x[i] > upperboundary[i] - width[i] + 0.00001`
I have to use something like 0.00001. Or should I define a constant like
`const double EPSILON = 0.00001`

Reply:
This "Magic Number" issue is fixed.